app/main.py

```diff
@@ -1,2 +1,110 @@
+from __future__ import annotations
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.size = 0
+        self.dictionary = [-1] * self.capacity
+
+    def hash_func(self, key: any) -> int:
+        return hash(key) % self.capacity
+
+    def resize_dict(self) -> None:
+        self.capacity = self.capacity * 2
+        temp_list = self.dictionary[:]
+        self.dictionary = [-1] * self.capacity
+        self.size = 0
+        for item in temp_list:
+            if item != -1:
+                self.__setitem__(item[0], item[2])
+
+    def __setitem__(self, key: any, value: any) -> None:
+        if self.size + 1 > self.capacity * 2 / 3:
+            self.resize_dict()
+        hash_for_key = self.hash_func(key)
+
+        if (self.dictionary[hash_for_key] == -1
+                or self.dictionary[hash_for_key] == "deleted"):
```

Comment:
because there can be a situation like:
dictionary[2] = 5 
then : del dictionary[2] -> dictionary["deleted"] (it is indexed by 2 in list)
and when we need to set again : dictionary[2] = 10, we need to check also if it is not deleted one

Reply:
In a way of adding a `deleted` flag you introduce complexity because there is a need to additionally handle `deleted` elements, at the same time there is no difference of `deleted` elements compared to `None` elements. Please, redefine the logic for setting and deleting an element to omit using the `deleted` keyword.